### PR TITLE
FM2-550 FHIR IG RelatedPerson Resource

### PR DESCRIPTION
**Pull request by k4pran**
Add implementation guide for `RelatedPerson` - https://openmrs.atlassian.net/browse/FM2-550

### CODE REVIEW

**Comment by dkayiwa** [comment]
@abertnamanya what do you think about this?

File: input/fsh/omrs-related-person.fsh [hunk 1]
```diff
@@ -0,0 +1,137 @@
+Profile: OMRSRelatedPerson
+Parent: RelatedPerson
+Id: omrs-related-person
+Title: "OpenMRS Related Person"
+Description: "A FHIR RelatedPerson as understood by OpenMRS Relationship"
+* id 1..1
+* patient 1..1
+* identifier 1..1
```
Comment (abertnamanya) [inline input/fsh/omrs-related-person.fsh:8]: The cardinality has to be one-to-many for example

File: input/fsh/omrs-related-person.fsh [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+* address 0..*
+* name 0..*
+* gender 0..1
+* birthDate 0..1
+* active 1..1
+* period 1..1
```
Comment (abertnamanya) [inline input/fsh/omrs-related-person.fsh:14]: Work on the cardinalities, Please look through this https://build.fhir.org/relatedperson.html It should be able to guide you.

File: input/fsh/omrs-related-person.fsh [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+* name.id = "915c9146-8bdb-4fad-bc51-4af10d041c86"
+* name.family = "John"
+* name.given[0] = "Smith"
+* gender = #male
+* birthDate = "2001-09-08"
+* active = true
+* period.start = "2020-01-01"
+* period.end = "2024-01-01"
+
+Mapping: RelatedPersonMapping
```
Comment (abertnamanya) [inline input/fsh/omrs-related-person.fsh:34]: Use this [test example](https://github.com/openmrs/openmrs-module-fhir2/blob/b20caa03498f860807f90660c63a7a9566bf6fee/api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImplTest.java#L102) to get the actual mappings

File: input/fsh/omrs-related-person.fsh [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+* active -> "voided"
+* name.family -> "Relationship.personA.names.family"
+* name.given -> "Relationship.personA.names.given"
+* gender -> "Relationship.personA.gender"
+* birthDate -> "Relationship.personA.birthdate"
+* patient -> "Relationship.personB"
+* period.start -> "Relationship.startDate"
+* period.end -> "Relationship.endDate"
+
+// Search parameters for OMRSRelatedPerson
```
Comment (abertnamanya) [inline input/fsh/omrs-related-person.fsh:53]: Extend the search parameters, The search parameters at the bottom of the [page ](https://build.fhir.org/relatedperson.html)should be able to guide you

**Review by abertnamanya — CHANGES_REQUESTED** [review]

File: input/fsh/omrs-related-person.fsh (hunk 3 above)
Comment (k4pran) [reply input/fsh/omrs-related-person.fsh:34]: Updated, I removed active from mappings as it seems fsh doesn't support conditional logic, I was using "voided" as some other .fsh did this so I can add back the mapping if you think it should map to voided. I also removed the URL from the identifier system and added missing address. 

I wasn't sure if we should map the patient by reference or not, something like

`* patient.reference -> "Relationship.personB.uuid"`

I can change it to this if needed.

I didn't change much else for mappings so please let me know if I have missed anything

File: input/fsh/omrs-related-person.fsh (hunk 2 above)
Comment (k4pran) [reply input/fsh/omrs-related-person.fsh]: Updated to match the official FHIR IG

File: input/fsh/omrs-related-person.fsh (hunk 4 above)
Comment (k4pran) [reply input/fsh/omrs-related-person.fsh:53]: Updated to add missing search params

File: input/fsh/omrs-related-person.fsh (hunk 1 above)
Comment (k4pran) [reply input/fsh/omrs-related-person.fsh]: Updated

**Pull request merged**